src/app/games/hive/HiveFailure.ts

```diff
@@ -0,0 +1,39 @@
+import { Localized } from 'src/app/utils/LocaleUtils';
+
+export class HiveFailure {
+
+    public static readonly CANNOT_DROP_PIECE_YOU_DONT_HAVE: Localized = () => `You cannot drop a piece that you do not have in your reserve`;
+
+    public static readonly QUEEN_BEE_CAN_ONLY_MOVE_TO_DIRECT_NEIGHBORS: Localized = () => $localize`The queen bee can only move to one of its direct neighbors.`;
+
+    public static readonly BEETLE_CAN_ONLY_MOVE_TO_DIRECT_NEIGHBORS: Localized = () => $localize`The beetle can only move to one of its direct neighbors.`;
+
+    public static readonly GRASSHOPPER_MUST_MOVE_IN_STRAIGHT_LINE: Localized = () => $localize`The grasshopper must move in a straight line.`;
+
+    public static readonly GRASSHOPPER_MUST_JUMP_OVER_PIECES: Localized = () => $localize`The grasshopper must jump over other pieces, without any empty spaces.`;
+
+    public static readonly SPIDER_MUST_MOVE_OF_3_NEIGHBORS: Localized = () => $localize`The spider must move of exactly three neighboring spaces.`;
+
+    public static readonly SPIDER_CAN_ONLY_MOVE_ON_EMPTY_SPACES: Localized = () => $localize`The spider can move only on empty spaces.`;
+
+    public static readonly SPIDER_CAN_ONLY_MOVE_WITH_DIRECT_CONTACT: Localized = () => $localize`The spider can only move around pieces that it is in direct contact with.`;
```

Comment:
erh, c'est la meilleure formulation ? j'ai l'impression d'avoir "que c'est que c'est qu'elle est contact avec" :D

Reply:
C'est chaud à bien formuler celle-ci, t'as une idée ?

Reply:
The spider can only move around pieces that are in direct contect with it.

Une option de spécifier la différence:
other piece must be physically connected to the piece only at the beginning and end of the move, the spider must actually mainly physical contact the whole movement, like if she was crawling.

Après, j'ai pas lu la formalution dans le tuto officiel.

Reply:
wait, t'as deux propositions:
  1. C'est la phrase actuelle avec pour différence "it is in direct contact with" -> "are in direct contact with it"; ok pour ça, j'ai changé accordément.
  2. tu proposes un message de toast de 3 lignes ? :D  (on est pas dans l'tuto ici !)

Reply:
Après euh faut prendre en compte que techniquement tu as les indicators qui te disent où tu peux cliquer, donc si tu cliques autre part ben tant pis !